pkg/neotest/coverage.go

```diff
@@ -0,0 +1,153 @@
+package neotest
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/nspcc-dev/neo-go/pkg/compiler"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+)
+
+var rawCoverage = make(map[scriptHash]*scriptRawCoverage)
+
+var enabled = false
+var coverProfile = ""
+
+type scriptHash = util.Uint160
+
+type scriptRawCoverage struct {
+	debugInfo      *compiler.DebugInfo
+	offsetsVisited []int
+}
+
+type coverBlock struct {
+	startLine uint // Line number for block start.
+	startCol  uint // Column number for block start.
+	endLine   uint // Line number for block end.
+	endCol    uint // Column number for block end.
+	stmts     uint // Number of statements included in this block.
+	counts    uint
+}
+
+type documentName = string
+
+func isCoverageEnabled() bool {
+	if enabled {
+		return true
+	}
+	const coverProfileFlag = "test.coverprofile"
+	flag.VisitAll(func(f *flag.Flag) {
+		if f.Name == coverProfileFlag && f.Value != nil {
+			enabled = true
+			coverProfile = f.Value.String()
+		}
+	})
+	if enabled {
+		// this is needed so go cover tool doesn't overwrite
+		// the file with our coverage when all tests are done
+		flag.Set(coverProfileFlag, "")
+	}
+	return enabled
+}
+
+func coverageHook() vm.OnExecHook {
+	return func(sh scriptHash, offset int, opcode opcode.Opcode) {
+		if cov, ok := rawCoverage[sh]; ok {
+			cov.offsetsVisited = append(cov.offsetsVisited, offset)
+		}
+	}
+}
+
+func reportCoverage() {
+	f, _ := os.Create(coverProfile)
```

Comment:
Ignoring an error is rarely a good idea. On the next line, we will panic with NPE, so why not have `panic(err)` before?
Same for `fmt.Fprintf` inside the `writeCoverageReport()`, it can error.
Another option is to write to a `bytes.Buffer` and then `os.WriteFile` once.